npc/re/jobs/2e/kagerou_oboro.txt

```diff
@@ -835,7 +835,7 @@ job_ko,81,124,4	script	Kuuga Gai#ko	730,{
 		setarray .@n[0],
 			1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,
 			26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50;
-		for (set .@i,getarraysize(.@n)-1; .@i>0; set .@i,.@i-1) {
+		for (set .@i,getarraysize(.@n)-1; .@i>1; set .@i,.@i-1) {
 			set .@rand, rand(.@i);
```

Comment:
I rather think that here it should not use .@i, but always the arraysize?

Reply:
I would replace the 'set's too

Reply:
I assume it's made to change the value at the given index and then the index is ignored in the next goto, so it can't be arraysize.
It looks correct but how about including the curent index in the randomization?
```
		for (set .@i,getarraysize(.@n)-1; .@i>0; set .@i,.@i-1) {
			set .@rand, rand(0,.@i);
```

Reply:
Honestly I'm not ready to replace those `set`s here 'cause I'll have to do that in the entire file too and why not properly refactor it further then... That's a trap I'm not going to fall into :D

Including the current index should be fine too, I think I'd do that then. If it happen to roll `.@rand` == `.@i` nothing bad will happen. 
I did it other way mostly because I wanted to minimize the changes and preserve the original logic as much as possible. The other thought was: starting the loop from size - 1 and going to > 0, not >= 0 as we do when browsing indices, is already strange enough to deserve more evident strangeness.